floodlight/io/catapultApi.py

```diff
@@ -0,0 +1,384 @@
+    if os.path.exists(endpoint):
+        # Walk through all directories and files in the given directory
+        for root, dirs, files in os.walk(endpoint):
+            for filename in files:
+                if filename.endswith(".json"):
+                    file_path = os.path.join(root, filename)
+                    
+                    # Load the JSON file and append the data to the list
+                    with open(file_path, "r") as json_file:
+                        data = json.load(json_file)
+                        response_data_json_list.append(data)
+    else:
+        raise FileNotFoundError(f"Directory {endpoint} does not exist.")
+    
+    return response_data_json_list
+
+## for testing end 
+
+# Custom Errors
+class APIRequestError(Exception):
+    """
+    Custom error class for API request failures.
+    """
+    def __init__(self, message: str):
+        super().__init__(message)
+
+def _getAllplayerFormActivity_response(baseUrl: str, api_token: str, activity_id: str) -> requests.Response:
+    """
+    Fetches data for all players involved in a specific activity.
+
+    Args:
+        baseUrl (str): The base URL for the API requests.
+        api_token (str): The API token for authentication.
+        activity_id (str): The unique identifier for the activity to fetch players from.
+
+    Returns:
+        requests.Response: The response object containing player data.
+
+    Raises:
+        APIRequestError: If the API request fails with a status code other than 200.
+    """
+    headers = {
+        "accept": "application/json",
+        "authorization": f"Bearer {api_token}",
+        "nulls": "0"
+    }
+    endpoint = f"activities/{activity_id}/athletes/"
+    response = requests.get(baseUrl + endpoint, headers=headers)
+    if response.status_code == 200:
+        return response
+    else:
+        raise APIRequestError(f"API request failed with status code {response.status_code}")
+
+def getResponse_data_json_list(baseUrl: str, api_token: str, activity_id: str) -> List[Dict]:
+    """
+    Retrieves sensor data for each player involved in a specific activity.
+
+    Args:
+        baseUrl (str): The base URL for the API requests.
+        api_token (str): The API token for authentication.
+        activity_id (str): The unique identifier for the activity.
+
+    Returns:
+        List[Dict]: A list of JSON data responses for each player, containing sensor data.
+
+    Raises:
+        APIRequestError: If any API request fails with a status code other than 200.
+    """
+    headers = {
+        "accept": "application/json",
+        "authorization": f"Bearer {api_token}",
+        "nulls": "0"
+    }
+    allPlayersFromActivity_response = _getAllplayerFormActivity_response(baseUrl, api_token, activity_id)
+    allPlayersFromActivity_JSON = allPlayersFromActivity_response.json()
+    playerIDsInActivity_list = list(set(i["id"] for i in allPlayersFromActivity_JSON))

```

Comment:
I think this line will lead to the players being in random order in each request since sets are unordered. Maybe just use `playerIDsInActivity.sorted()` on the set afterwards. You are converting the set to a list anyways.

Reply:
Done